File: app/src/main/java/com/synapse/social/studioasinc/widget/ZoomImageViewLib/animation/SpringInterpolator.java
package com.synapse.social.studioasinc.widget.ZoomImageViewLib.animation;

import android.view.animation.Interpolator;

public class SpringInterpolator implements Interpolator {
    private float factor;

    public SpringInterpolator(float factor2) {
        this.factor = factor2;
    }

    public float getInterpolation(float input) {
        return (float) ((Math.pow(2.0d, (double) (-10.0f * input)) * Math.sin((((double) (input - (this.factor / 4.0f))) * 6.283185307179586d) / ((double) this.factor))) + 1.0d);
    }
}

File: app/src/main/java/com/synapse/social/studioasinc/widget/ZoomImageViewLib/animation/SpringInterpolator.kt
package com.synapse.social.studioasinc.widget.ZoomImageViewLib.animation

import android.view.animation.Interpolator
import kotlin.math.pow
import kotlin.math.sin

class SpringInterpolator(private val factor: Float) : Interpolator {
    override fun getInterpolation(input: Float): Float {
        return (2.0.pow((-10.0f * input).toDouble()) * sin(((input - factor / 4.0f) * 6.283185307179586).toDouble() / factor) + 1.0).toFloat()
    }
}
